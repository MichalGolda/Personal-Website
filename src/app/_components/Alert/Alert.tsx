export type AlertProps = {
  className?: string;
  message: string;
};

const Alert: React.FC<AlertProps> = ({ className, message }) => {
  return (
    <div
      className={`w-full p-4 text-white top-0 left-0 font-bold text-center fixed z-[999]  ${className}`}
    >
      <p>{message}</p>
    </div>
  );
};

export default Alert;
