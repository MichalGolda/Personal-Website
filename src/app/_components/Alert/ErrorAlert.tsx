import Alert from "./Alert";
import type { AlertProps } from "./Alert";

export default function ErrorAlert({ message }: AlertProps) {
  return <Alert className="bg-error" message={message} />;
}
