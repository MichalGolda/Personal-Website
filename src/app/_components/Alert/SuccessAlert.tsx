import Alert from "./Alert";
import type { AlertProps } from "./Alert";

export default function SuccessAlert({ message }: AlertProps) {
  return <Alert className="bg-success" message={message} />;
}
