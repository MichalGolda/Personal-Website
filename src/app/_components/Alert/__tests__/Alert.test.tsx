import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import Alert from "../Alert"

describe("Alert component", () => {
    it("renders without errors", () => {
        render(<Alert message="test message" />)

        const alert = screen.getByTestId("alert")
        const alertMessage = screen.getByText("test message")

        expect(alert).toBeInTheDocument()
        expect(alertMessage).toBeInTheDocument()
    })

    it("renders unchanged", () => {
        const { container } = render(<Alert message="test message" />)
        
        expect(container).toMatchSnapshot()
    })
})
