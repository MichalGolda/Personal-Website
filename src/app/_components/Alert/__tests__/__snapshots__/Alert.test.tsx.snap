// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component renders unchanged 1`] = `
<div>
  <div
    class="w-full p-4 text-white bottom-0 left-0 font-bold text-sm md:text-base text-center fixed z-[999]  undefined"
    data-testid="alert"
  >
    <p>
      test message
    </p>
  </div>
</div>
`;
